### PR TITLE
Babel googleanalytics

### PR DESCRIPTION
**Pull request by armoj**
Adds some custom event tracking to the user-facing application.
* Custom scroll threshold event tracking
* Detail element interaction event tracking

Build to work in the current list of GOVUK supported browsers/devices (listed here: https://www.gov.uk/service-manual/technology/designing-for-different-browsers-and-devices#browsers-to-test-in). At time of development, the support is as follows:
* IE - 11+ (windows)
* Edge - latest (windows)
* Chrome - latest (windows/MacOS/iOS/Android)
* FF - latest (windows/MacOS)
* Safari - 12+ (MacOS/iOS)

I have also tested it in FF on Android devices.

The majority of the testing was carried out using BrowserStack. I have also tested on my local machine for all combinations that are satisfied with my local setup (i.e. Windows 10 FF, Chrome, Edge, IE11)

This functionality works in a large number of unofficially supported browsers. For example, the functionality introduced in this PR works in:
* IE 8/9/10 (windows)
* Chrome 60+ (windows)
* FF 40, 50+ (windows)

In addition to all the above, I have tested at least one version lower (than officially supported) for each of the supported combinations.

### CODE REVIEW

File: index/start-page.njk [hunk 1]
```diff
@@ -60,7 +60,7 @@
            <li>in England and Wales <a href="https://www.victimandwitnessinformation.org.uk/">visit the Victim and Witness Information website</a></li>
            <li>in Scotland <a href="https://www.mygov.scot/victim-witness-support/">visit the mygov.scot website</a></li>
         </ul>
-        <p class="govuk-body">You do not need a legal representative to make an application.</p>
+        <p class="govuk-body ga-event ga-event--scrollto" ga-event-action="scrollTo" ga-event-label="bottom" ga-event-category="start-page">You do not need a legal representative to make an application.</p>
```
Comment (sinclairs) [inline index/start-page.njk]: Do these align to? https://support.google.com/analytics/answer/1033068#Anatomy

Some of these could be set with non-custom attributes e.g. using
- The scroll handler
- window.location or something? 

The custom attributes should be prefixed with "data-"

File: index/start-page.njk [hunk 2]
```diff
@@ -2,7 +2,7 @@
 {% from "components/button/macro.njk" import govukButton %}
 
 {% block content %}
-<main role="main" class="govuk-main-wrapper">
+<main role="main" class="govuk-main-wrapper ga-event ga-event--scrollthreshold">
```
[the remark this answers is not in the document]
Comment (armoj) [reply index/start-page.njk]: I'm happy for a first pass at it. Just want to get it working solid now and then we can rethink it.

File: package.json [hunk 3]
```diff
@@ -3,21 +3,27 @@
...
+        "copy:css": "rm -rf public/dist/css && mkdir -p public/dist/css && cp -r ./src/modules/**/*.css public/dist/css/ || :",
+        "copy:js": "rm -rf public/dist/js && mkdir -p public/dist/js && cp -r ./src/modules/**/vendor/*.js public/dist/js/ || :"
     },
     "dependencies": {
         "@babel/core": "^7.7.4",
         "@babel/preset-env": "^7.7.4",
         "babel-loader": "^8.0.6",
         "client-sessions": "github:CriminalInjuriesCompensationAuthority/node-client-sessions#v0.9.0",
         "cookie-parser": "~1.4.3",
+        "core-js": "^3.5.0",
```
[the remark this answers is not in the document]
Comment (armoj) [reply package.json]: No, it needs to be in "dependencies" for it to build to production

File: src/modules/ga/index.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            }
+        });
+    }
+
+    return Object.freeze({
+        setUpGAEventTracking
+    });
+}
+
+// eslint-disable-next-line import/prefer-default-export
```
[the remark this answers is not in the document]
Comment (sinclairs) [reply src/modules/ga/index.js]: what was the outcome of this?

**Comment by sinclairs** [comment]
> 
> 
> Can you stick on a description of the PR and what testing has been done

Not sure If I'm missing it but can you add a description in e.g.

* What the PR does
* What browsers you expect it to work in
* What testing has been done to ensure it works in those browsers
* What happens in unsupported browsers

File: src/modules/ga/index.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (armoj) [reply src/modules/ga/index.js]: this has been updated since the start of this conversation! eslint comment no longer needed

**Pull request merged**